src/main/kotlin/build/buf/intellij/fixes/IgnoreBufIssueQuickFix.kt

```diff
@@ -47,26 +50,56 @@ class IgnoreBufIssueQuickFix(private val type: String) : BaseIntentionAction() {
         val yamlRootMapping = bufConfigFile.children.firstOrNull()?.children?.find {
             it is YAMLMapping
         } as? YAMLMapping ?: return
-        val lintNode = yamlRootMapping.getKeyValueByKey("lint")
-        if (lintNode is YAMLMapping && lintNode.getKeyValueByKey("allow_comment_ignores") != null) {
+        val version = yamlRootMapping.getKeyValueByKey("version")?.valueText ?: return
+        when (version) {
+            "v2" -> enableCommentIgnoresV2(yamlRootMapping)
+            "v1", "v1beta1" -> enableCommentIgnoresV1(project, bufConfigFile, yamlRootMapping)
+            else -> return
+        }
+    }
+
+    private fun enableCommentIgnoresV2(yamlRootMapping: YAMLMapping) {
+        // Disable on all modules if overridden there.
+        val modules = yamlRootMapping.getKeyValueByKey("modules")?.value as? YAMLSequence
+        if (modules != null) {
+            for (module in modules.items) {
+                val moduleMapping = module?.value as? YAMLMapping ?: continue
+                enableCommentIgnoresV2(moduleMapping)
+            }
+        }
+        val lintNode = yamlRootMapping.getKeyValueByKey("lint") ?: return
+        val lintNodeValue = lintNode.value as YAMLMapping? ?: return
+        val disallowCommentIgnoresNode = lintNodeValue.getKeyValueByKey("disallow_comment_ignores") ?: return
+        val disallowCommentIgnores = disallowCommentIgnoresNode.value as? YAMLScalar ?: return
+        if (disallowCommentIgnores.textValue != "true") {
             return
         }
+        disallowCommentIgnoresNode.parentMapping?.deleteKeyValue(disallowCommentIgnoresNode)
+        if (lintNodeValue.keyValues.isEmpty()) {
+            yamlRootMapping.deleteKeyValue(lintNode)
+        }
+    }
+
+    private fun enableCommentIgnoresV1(project: Project, bufConfigFile: YAMLFile, yamlRootMapping: YAMLMapping) {
         val codeStyleManager = CodeStyleManager.getInstance(project)
+        val lintNode = yamlRootMapping.getKeyValueByKey("lint")
         if (lintNode == null) {
             val generateNode = PsiFileFactory.getInstance(project)
                 .createFileFromText(YAMLLanguage.INSTANCE, "\nlint:\n\tallow_comment_ignores: true")
                 .children.first()
             yamlRootMapping.add(generateNode.firstChild)
             yamlRootMapping.add(generateNode.lastChild)
             codeStyleManager.reformatText(bufConfigFile, listOf(yamlRootMapping.textRange))
-        } else {
-            val generateNode = PsiFileFactory.getInstance(project)
-                .createFileFromText(YAMLLanguage.INSTANCE, "\nallow_comment_ignores: true")
```

Comment:
This just checked if it wasn't null before and would add this setting set to true, however that could lead to duplicate keys being added to the config. Fixed this and added tests for more variations.